### ISSUE TITLE
FAQ: What is a Kubernetes cluster?

### ISSUE THREAD

**Issue by lxie123** [kube-academy/backlog#33]
FAQ: What is a Kubernetes cluster?

**Comment by jonasmit** [comment]
draft expected of script friday sept 10

**Comment by jonasmit** [comment]
Syeda will complete after the 28th to study for the CKA/CKAD Exam...

**Comment by lxie123** [comment]
Video available for review. 
https://drive.google.com/drive/u/0/folders/13bNlIOCvWwdz7dgaFaJtzxOvLNWr7wBF

Looks good to me.

**Comment by eitansuez** [comment]
1. not sure i agree with the statement "must have at least one control plane node and one worker node to be operational."
2. in my opinion, a description of how k8s clusters work that does not also discuss control loops and the components of the control plane (the kube api server, kube scheduler, and kube controller manager) provides at best an incomplete picture and understanding of the mechanism.
3. i'm not sure what it means to say that the control plane node scales applications.  multiple components are involved in the process, including the kubelet which runs on worker nodes.
4. i don't think this statement is true: "when traffic is high the control plane node will deploy more instances to balance traffic."  this is not going to happen unless you configure pod autoscaling.
i worry that perhaps this video provides too simplistic an explanation of a kubernetes cluster.
i wonder whether this topic is perhaps not the best candidate for a 2-3 minute faq, in that, in order to properly explain how a k8s cluster functions, one would perhaps require more time?

**Comment by djschny** [comment]
After reviewing the video, really awesome job on clarity and delivery of the content. I made note of the following items:

1. The video box being in the bottom-right corner feels unnatural compared to the top-right which tends to be defacto and matches the other KA videos. It also covers up the slide number which is nice for reference.
1. The video is 2:35 total but of that about 30 seconds of the video is intro and talking about what is going to be talked about in the video. For short videos like this, taking time for agenda/outline/preview discussion is not really needed and removing helps allow the videos to be to the point and focus on content.
1. The lines on the `Kubernetes Cluster` slide are not aligned/proper.
1. As mentioned above the statement `must have at least one control plane node and one worker node to be operational` is not correct and yes should be removed IMO
1. On the `Control Plane Node` slide there is a statement `It does this by communicating with the worker nodes to know it's status` is incorrect and should be reworked or removed. The nodes report their state to the API server, there is not communication outbound or to the worker nodes which the statement infers.

I like the video overall but the two statements mentioned above really do need to be corrected. If it's going to be recorded again to fix those statements then I would also suggest addressing the other minor items at the same time.

**Comment by shaqster** [comment]
Thanks so much for the feedback! I'm on an engagement now. I will work on redoing the video soon

**Comment by lxie123** [comment]
waiting on @shaqster to have time to make final edits.

**Comment by jonasmit** [comment]
@lxie123 - mentioend @shaqster should have more time after current engagement which wraps up around the 19th

**Comment by shaqster** [comment]
will try to get the script and video done by 12/13

**Comment by lxie123** [comment]
@lxie123 will check with Syeda about 12/13, Will do it ourselves if she can't make that date.

**Comment by lxie123** [comment]
@eitansuez @djschny Please review Syeda's changes per your feedback and let her know if she can re record.
https://docs.google.com/document/d/1S65vtbJLmwjsS15YYTG2lqw3qntRMB8n-6X3GhCB6Kg/edit

**Comment by eitansuez** [comment]
the script looks good to me. thanks.

**Comment by djschny** [comment]
Updated recording done last night as well - https://drive.google.com/drive/u/1/folders/13bNlIOCvWwdz7dgaFaJtzxOvLNWr7wBF

**Comment by eitansuez** [comment]
i have reviewed the recording and given syeda more feedback.  let's discuss at standup.

**Comment by eitansuez** [comment]
at standup on dec 14, we decided that syeda's recording is good / has passed technical review.
i was asked to capture my last piece of feedback to syeda in this github issue.  here it is:
-------
hi syeda,
i just watched the recording.  it's good.  i wonder whether i can push you to do one more round?
i can't quite put my finger on it but i think the narrative can be refined further to make that matter really clear to the audience.
i know you have other responsibilities besides this, so feel free to push back and we can just go with this recording.
i asked myself: fundamentally what is the difference between the control plane and other vm's and can that difference be explained to provide insight into how the thing functions, without having to go into too much detail?
i'd like to share with you an attempt (see below) i just made at the script.  i don't want you to use that script verbatim.  rather, it's an example of a structure for the narrative that i think achieves the above (focuses on the difference between the two types of nodes without going into too much detail).
let me know what you think and what you'd like to do.
---
What is a kubernetes cluster?
A kubernetes cluster is a collection of virtual machines that provide a solution to the problem of running workloads (applications) in containers -- this is often referred to as "container orchestration."
Specific VMs are designated as "Control Plane" VMs (or nodes) while others are "Worker VMs."
The designation "Control Plane" means that those VMs are reserved for running processes involved with controlling how the cluster functions.
In other words, most of the machinery of kubernetes runs on control plane nodes.
The applications that developers deploy to a kubernetes cluster do not run on control plane nodes, they run on the worker nodes.
So, the worker nodes are the pool of compute resources reserved for running workloads.
To summarize, if I am a developer with access to a Kubernetes cluster, I communicate with a process running on the *control plane* to request, say, the deployment of my app to the cluster.  My app ends up running (in a container) on one or more (if it's a scaled deployment) of the *worker* VMs.
There's more to the story, for example, how does Kubernetes arrange for these workloads to run in containers?  that's a question we will tackle in a future follow-on lesson.
